src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,75 @@
+package duke;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+import duke.exception.SaveException;
+import duke.exception.SaveException.SaveExceptionType;
+
+public class Storage {
+    protected String filepath;
+
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    protected TaskList load() throws IOException, SaveException {
+        TaskList tasks = null;
+        FileInputStream fs = null;
+        ObjectInputStream os = null;
+        try {
+            fs = new FileInputStream(this.filepath);
+            os = new ObjectInputStream(fs);
+            Object obj = os.readObject();
+
+            if (obj instanceof TaskList) {
+                tasks = (TaskList) obj;
+                tasks.setStorage(this);
+            } else {
+                throw new SaveException(SaveExceptionType.INVALID_SAVE);
+            }
+        } catch (FileNotFoundException e) {
+            throw new SaveException(SaveExceptionType.NO_SAVE, e);
+        } catch (ClassNotFoundException e) {
+            throw new SaveException(SaveExceptionType.INVALID_SAVE, e);
+        } finally {
+            // Close all open file and object handles
+            if (os != null) {
+                os.close();
+            }
+            if (fs != null) {
+                fs.close();
+            }
```

Comment:
You can consider SLAP-ing this for better reusability since you're using it in the following `save` method as well

Reply:
They are actually different as this is the input stream and the `save` ones are output stream. Probably because of the inappropriate naming. It has been renamed more clearly in the latest commit.